tests/unit/test_navigation_table.py

```diff
@@ -0,0 +1,241 @@
+        ),
+        pytest.param("|1|a |[b]()|", (1, "a", ("b", None)), id="simple space after second column"),
+        pytest.param("|1|a| [b]()|", (1, "a", ("b", None)), id="simple space before third column"),
+        pytest.param("|1|a|[b]() |", (1, "a", ("b", None)), id="simple space after third column"),
+        pytest.param("|1|a|[b]()| ", (1, "a", ("b", None)), id="simple trailing space"),
+        pytest.param("|12|a|[b]()|", (12, "a", ("b", None)), id="first column multiple digits"),
+        pytest.param(
+            "|1|az|[b]()|", (1, "az", ("b", None)), id="second column multiple characters"
+        ),
+        pytest.param("|1|A|[b]()|", (1, "A", ("b", None)), id="second column upper case"),
+        pytest.param("|1|2|[b]()|", (1, "2", ("b", None)), id="second column digit"),
+        pytest.param("|1|_|[b]()|", (1, "_", ("b", None)), id="second column underscore"),
+        pytest.param("|1|-|[b]()|", (1, "-", ("b", None)), id="second column dash"),
+        pytest.param(
+            "|1|a|[bz]()|", (1, "a", ("bz", None)), id="third column title multiple characters"
+        ),
+        pytest.param(
+            "|1|a|[B]()|", (1, "a", ("B", None)), id="third column title capital character"
+        ),
+        pytest.param("|1|a|[2]()|", (1, "a", ("2", None)), id="third column title digit"),
+        pytest.param("|1|a|[_]()|", (1, "a", ("_", None)), id="third column title underscore"),
+        pytest.param("|1|a|[-]()|", (1, "a", ("-", None)), id="third column title dash"),
+        pytest.param("|1|a|[ ]()|", (1, "a", (" ", None)), id="third column title space"),
+        pytest.param("|1|a|[ b]()|", (1, "a", ("b", None)), id="third column title space before"),
+        pytest.param("|1|a|[b ]()|", (1, "a", ("b", None)), id="third column title space after"),
+        pytest.param(
+            "|1|a|[b c]()|", (1, "a", ("b c", None)), id="third column title embedded space"
+        ),
+        pytest.param("|1|a|[b](c)|", (1, "a", ("b", "c")), id="third column link defined"),
+        pytest.param(
+            "|1|a|[b] (c)|",
+            (1, "a", ("b", "c")),
+            id="third column white spave between title and link",
+        ),
+        pytest.param(
+            "|1|a|[b]( c)|", (1, "a", ("b", "c")), id="third column link whitespace before"
+        ),
+        pytest.param(
+            "|1|a|[b](c )|", (1, "a", ("b", "c")), id="third column link whitespace after"
+        ),
+        pytest.param(
+            "|1|a|[b](cd)|", (1, "a", ("b", "cd")), id="third column link multiple characters"
+        ),
+        pytest.param("|1|a|[b](C)|", (1, "a", ("b", "C")), id="third column link upper case"),
+        pytest.param("|1|a|[b](2)|", (1, "a", ("b", "2")), id="third column link digit"),
+        pytest.param("|1|a|[b](/)|", (1, "a", ("b", "/")), id="third column link forward slash"),
+        pytest.param("|1|a|[b](-)|", (1, "a", ("b", "-")), id="third column link dash"),
```

Comment:
URL could contain `:` and `.` should be part of the test I think, unless it'll always be relative links ?

Reply:
Yes always relative link generated by discourse which are of the form `/t/<slug>/<topic-id>`